fineract-provider/src/test/java/org/apache/fineract/infrastructure/jobs/filter/LoanCOBApiFilterTest.java

```diff
@@ -299,16 +297,15 @@ void shouldRunInlineCOBAndProceedWhenLoanIsBehind() throws ServletException, IOE
 
         LoanIdAndLastClosedBusinessDate result = mock(LoanIdAndLastClosedBusinessDate.class);
         given(result.getId()).willReturn(2L);
+        given(result.getLastClosedBusinessDate()).willReturn(businessDate.minusDays(2));
         given(request.getPathInfo()).willReturn("/v1/loans/2?command=approve");
         given(request.getMethod()).willReturn(HTTPMethods.POST.value());
         given(loanAccountLockService.isLoanHardLocked(2L)).willReturn(false);
-        given(loanAccountLockService.isLoanSoftLocked(2L)).willReturn(true);
         given(fineractProperties.getQuery()).willReturn(fineractQueryProperties);
         given(fineractQueryProperties.getInClauseParameterSizeLimit()).willReturn(65000);
-        given(loanRepository.findAllNonClosedLoansBehindOrNullByLoanIds(
+        given(loanRepository.findAllNonClosedLoansBehindByLoanIds(
```

Comment:
why did you change this?

Reply:
LoanCOBApiFilter isLoanBehind Method calls findAllNonClosedLoansBehindByLoanIds and not findAllNonClosedLoansBehindOrNullByLoanIds. 
The test was passing earlier as isLoanSoftLock and isLoanBehind were in OR condition ((isLoanSoftLocked(loanIds) || isLoanBehind(loanIds)) and test was setting softlock so isLoanBehind condition was not getting checked.
Once isSoftLock check was removed test started failing...

Reply:
The inline COB job in the other hand is using the `findAllNonClosedLoansBehindOrNullByLoanIds` it should have been called...

I guess we just need to check further this test case was about to check what particular test case.